src/gridfs-stream/download.ts

```diff
@@ -237,28 +239,28 @@ function doRead(stream: GridFSBucketReadStream): void {
     let errmsg: string;
     if (doc.n > expectedN) {
       errmsg = 'ChunkIsMissing: Got unexpected n: ' + doc.n + ', expected: ' + expectedN;
-      return __handleError(stream, new Error(errmsg));
+      return __handleError(stream, new MongoDriverError(errmsg));
     }
 
     if (doc.n < expectedN) {
       errmsg = 'ExtraChunk: Got unexpected n: ' + doc.n + ', expected: ' + expectedN;
-      return __handleError(stream, new Error(errmsg));
+      return __handleError(stream, new MongoDriverError(errmsg));
     }
 
     let buf = Buffer.isBuffer(doc.data) ? doc.data : doc.data.buffer;
 
     if (buf.byteLength !== expectedLength) {
       if (bytesRemaining <= 0) {
         errmsg = 'ExtraChunk: Got unexpected n: ' + doc.n;
-        return __handleError(stream, new Error(errmsg));
+        return __handleError(stream, new MongoDriverError(errmsg));
       }
 
       errmsg =
```

Comment:
There's a lot of concat strings here that could template strings, its understandable if you wouldn't like to add on this PR further with some code clean up, but figure I'd suggest we do it here anyway. Feel free to resolve without changes.

Reply:
resolving for intern work